src/client/datascience/jupyter/jupyterSession.ts

```diff
@@ -82,40 +72,34 @@ export class JupyterSession extends BaseJupyterSession {
     }
 
     public async createNewKernelSession(
-        resource: Resource,
-        kernelConnection: KernelConnectionMetadata | undefined,
-        timeoutMS: number,
         cancelToken?: CancellationToken,
         disableUI?: boolean
     ): Promise<ISessionWithSocket> {
         let newSession: ISessionWithSocket | undefined;
-
-        // update resource as we know it now.
-        this.resource = resource;
```

Comment:
This is now gone, this is no longer required, in the ctor we define this property, hence its never nullable.